ui-admin/src/study/participants/enrolleeView/EnrolleeTimeline.tsx

```diff
@@ -0,0 +1,147 @@
+import { instantToDefaultString } from 'util/timeUtils'
+import NotificationConfigTypeDisplay from '../../notifications/NotifcationConfigTypeDisplay'
+import { Link } from 'react-router-dom'
+
+const eventClassToHumanReadable = (eventClass: string): string => {
+  const output = eventClass
+    .split('_')
+    .map(str => str.toLowerCase())
+    .join(' ')
+
+  // capitalize first letter of output
+  return output[0].toUpperCase() + output.substring(1)
+}
+
+type EventNotificationTableEntry = {
+  id: string,
+  createdAt: number,
+  deliveryStatus?: string,
+  deliveryType?: string,
+  notificationConfig?: NotificationConfig,
+  notificationConfigId?: string,
+  sentTo?: string,
+  eventClass?: string,
+}
+
+
+/** loads the list of notifications and events for a given enrollee and displays them in the UI */
+export default function EnrolleeTimeline({ enrollee, studyEnvContext }:
+                                           { enrollee: Enrollee, studyEnvContext: StudyEnvContextT }) {
+  const { currentEnv, study, portal, currentEnvPath } = studyEnvContext
+  const [notifications, setNotifications] = useState<Notification[]>([])
+  const [events, setEvents] = useState<Event[]>([])
+  const [tableData, setTableData] = useState<EventNotificationTableEntry[]>([])
+  const [isLoading, setIsLoading] = useState(true)
+
+  /** matches each notification to a corresponding config by id */
+  function attachConfigsToNotifications(rawNotifications: Notification[]) {
+    rawNotifications.forEach(notification => {
+      notification.notificationConfig = currentEnv.notificationConfigs
+        .find(config => config.id === notification.notificationConfigId)
+    })
+  }
+
+  const notificationToTableEntry = (val: Notification): EventNotificationTableEntry => {
+    return {
+      id: val.id,
+      deliveryType: val.deliveryType,
+      deliveryStatus: val.deliveryStatus,
+      notificationConfig: val.notificationConfig,
+      notificationConfigId: val.notificationConfigId,
+      createdAt: val.createdAt
+    }
+  }
+
+  const eventToTableEntry = (val: Event): EventNotificationTableEntry => {
+    return {
+      id: val.id,
+      createdAt: val.createdAt,
+      eventClass: val.eventClass
+    }
+  }
+
+  useEffect(() => {
+    Api.fetchEnrolleeNotifications(portal.shortcode, study.shortcode, currentEnv.environmentName, enrollee.shortcode)
+      .then(response => {
+        attachConfigsToNotifications(response)
+        setNotifications(response)
+        Api.fetchEnrolleeEvents(portal.shortcode, study.shortcode, currentEnv.environmentName, enrollee.shortcode)
+          .then(response => {
+            setEvents(response)
+            setIsLoading(false)
+          })
+      })
+  }, [enrollee.shortcode])
+
+  useEffect(() => {
+    const newTableData: EventNotificationTableEntry[] = []
+
+    newTableData.push(...notifications.map(notificationToTableEntry))
+    newTableData.push(...events.map(eventToTableEntry))
```

Comment:
```suggestion
    const newTableData = [...(notifications as EventNotificationTableEntry[]), ...(events as EventNotificationTableEntry[])]
```
And if you change the type declaration of EventNotificationTableEntry to be Event | Notification you might not even need the `as` casting